splunk-forwarder/main.tf

```diff
@@ -0,0 +1,111 @@
+resource "random_string" "unique_identfier" {
+  length  = 8
+  special = false
+}
+
+resource "aws_iam_role" "lambda_role" {
+  name = "lambda_role_${random_string.unique_identfier.result}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": "sdsd"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "allow_cloudwatch" {
+  name        = "cw_for_lambda_${random_string.unique_identfier.result}"
+  description = "Lets the Lambda function write to cloudwatch"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents",
+        "logs:DescribeLogStreams"
+    ],
+      "Resource": [
+        "arn:aws:logs:*:*:*"
+    ]
+  }
+ ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "allow_s3_puts" {
+  name        = "s3_for_lambda_${random_string.unique_identfier.result}"
+  description = "Lets the Lambda function write to a any bucket"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "s3:*"
```

Comment:
This privilege is too wide? I think it would be nice to restrict it a bit here, so that it's clearer what this function actually does?

Reply:
Absolutely